source/core.py

```diff
@@ -262,6 +262,55 @@ def onEndSession(evt):
 	import windowUtils
 	class MessageWindow(windowUtils.CustomWindow):
 		className = u"wxWindowClassNR"
+		oldBatteryStatus = None
+		hasBeenInitialized = False
+		#Just define these constants here, so we don't have to import win32con
+		WM_POWERBROADCAST = 0x218
+		WM_DISPLAYCHANGE = 0x7e
+		PBT_APMPOWERSTATUSCHANGE = 0xA
+		UNKNOWN_BATTERY_STATUS = 0xFF
+		AC_ONLINE = 0X1
+		NO_SYSTEM_BATTERY = 0X80
+
+		def __init__(self, windowName=None):
+			super(MessageWindow, self).__init__(windowName)
+			self.handlePowerStatusChange(initializing=True)
+
+		def windowProc(self, hwnd, msg, wParam, lParam):
+			import ui
+			import winUser
+			if msg == self.WM_POWERBROADCAST and wParam == self.PBT_APMPOWERSTATUSCHANGE:
+				self.handlePowerStatusChange()
+			elif msg == self.WM_DISPLAYCHANGE:

```

Comment:
I normally wouldn't suggest this, but because this is in the middle of a very general window procedure which will probably get extended to support all sorts of things, I'd prefer the code inside this block to handle orientation was split into its own method too.